clientd3d/d3drender_skybox.c

```diff
@@ -137,15 +128,18 @@ static unsigned char gSkyboxBGRA[] =
 // Interfaces
 
 static void D3DRenderBackgroundsLoad(char* pFilename, int index);
-static void D3DRenderBackgroundSet(ID background);
+static bool D3DRenderBackgroundSet(ID background);
 
 // Implementations
 
 /**
-* Initialize the skybox rendering process.
+* Initialize the skybox with some rendering params.
+* Return true the background has changed and false otherwise.
+* If the background has changed, the caller should trigger a redraw all.
 */
-void D3DRenderSkyBoxBegin()
+bool D3DRenderSkyBoxBegin(SkyboxRenderParams& params)
 {
+	skyboxRenderParams = &params;
```

Comment:
Super dangerous / will crash: taking the address of a stack variable and storing it as a global variable.  Would be much, much better to pass this to functions that need it and remove the global variable.